### PR TITLE
Test on Java 17

### PR DESCRIPTION
**Pull request by sideeffffect**
Analogous to https://github.com/zio/zio/pull/5357
closes https://github.com/zio/zio-prelude/pull/686

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,7 +3,6 @@ name: CI
 env:
   JDK_JAVA_OPTIONS: -XX:+PrintCommandLineFlags -XX:MaxMetaspaceSize=4G -Xmx8G -Xss8M # JDK_JAVA_OPTIONS is _the_ env. variable to use for modern Java
   JVM_OPTS: -XX:+PrintCommandLineFlags -XX:MaxMetaspaceSize=4G -Xmx8G -Xss8M # for Java 8 only (sadly, it is not modern enough for JDK_JAVA_OPTIONS)
-  SBT_OPTS: -XX:+PrintCommandLineFlags -XX:MaxMetaspaceSize=4G -Xmx8G -Xss8M
```
Comment (sideeffffect) [inline .github/workflows/ci.yml:6]: My hunch is that this is not needed when we run `./sbt` (instead of `sbt`)

**Review by adamgfraser — APPROVED** [review]
Good to go when conflicts are resolved!

**Comment by sideeffffect** [comment]
Seriously :disappointed: what is causing these stalled builds? Is it GitHub Actions? Or is it ZIO itself :scream_cat:

**Comment by sideeffffect** [comment]
@adamgfraser @mijicd :pray:

**Comment by adamgfraser** [comment]
@sideeffffect Thank you! 🙏

**Pull request merged**